synapse/handlers/register.py

```diff
@@ -361,6 +362,59 @@ def _submit_captcha(self, ip_addr, private_key, challenge, response):
         )
         defer.returnValue(data)
 
+    @defer.inlineCallbacks
+    def create_user(self, localpart, displayname, duration_seconds):
+        """Creates a new user or returns an access token for an existing one
+
+        Args:
+            localpart : The local part of the user ID to register. If None,
+              one will be randomly generated.
+        Returns:
+            A tuple of (user_id, access_token).
+        Raises:
+            RegistrationError if there was a problem registering.
+        """
+        yield run_on_reactor()
+
+        if localpart is None:
+            raise SynapseError(400, "Request must include user id")
+
+        need_register = True
+
+        try:
+            yield self.check_username(localpart)
+        except SynapseError as e:
+            if e.errcode == Codes.USER_IN_USE:
+                need_register = False
+            else:
+                raise
+
+        user = UserID(localpart, self.hs.hostname)
+        user_id = user.to_string()
+        auth_handler = AuthHandler(self.hs)
```

Comment:
`self.hs.get_handlers().auth_handler` is a premade AuthHandler :)

So just: `auth_handler = self.hs.get_handlers().auth_handler`